pilot/helpers/agents/SpecWriter.py

```diff
@@ -0,0 +1,71 @@
+from helpers.AgentConvo import AgentConvo
+from helpers.Agent import Agent
+from utils.files import count_lines_of_code
+from utils.style import color_green_bold, color_yellow_bold
+from prompts.prompts import ask_user
+
+INITIAL_PROJECT_HOWTO_URL = "https://github.com/Pythagora-io/gpt-pilot/wiki/How-to-write-a-good-initial-project-description"
+
+class SpecWriter(Agent):
+    def __init__(self, project):
+        super().__init__('spec_writer', project)
+        self.save_dev_steps = True
+
+    def analyze_project(self, initial_prompt):
+        msg = (
+            "Your project description seems a bit short. "
+            "The better you can describe the project, the better GPT Pilot will understand what you'd like to build.\n\n"
+            f"Here are some tips on how to better describe the project: {INITIAL_PROJECT_HOWTO_URL}\n\n"
+        )
+        print(color_yellow_bold(msg))
+        print(color_green_bold("Let's start by refining your project idea:"))
+
+        convo = AgentConvo(self)
+        convo.construct_and_add_message_from_prompt('spec_writer/ask_questions.prompt', {})
+
+        user_response = initial_prompt
+        while True:
```

Comment:
I think user should be able to stop this loop. It asked me 20 questions and i stopped reading after 4 🤷‍♂️

Reply:
![Screenshot from 2024-02-16 18-14-10](https://github.com/Pythagora-io/gpt-pilot/assets/3362/600bcfbf-0618-4536-a184-5661ad4d8e7c)

Trace data example:
```js
  {
    _id: ObjectId('65d01691b98b13547056d737'),
    pathId: 'telemetry-senko-test',
    event: 'trace-spec-writer-questions',
    data: {
      initial_prompt_length: 15,
      num_questions: 2,
      final_prompt_length: 3904,
      skipped: true
    },
    createdAt: ISODate('2024-02-17T02:14:41.932Z'),
    updatedAt: ISODate('2024-02-17T02:14:41.932Z'),
  }
```